silkworm/db/snapshots/index/bloom_filter.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include "bloom_filter.hpp"
+
+#include <numbers>
+#include <random>
+#include <sstream>
+#include <string_view>
+
+#include <openssl/evp.h>
```

Comment:
should add OpenSSL::Crypto into snapshots/CMakeLists.txt (see how it is done in sentry/CMakeLists.txt)